File: src/harpy/ast/statements/static_variable_declaration.py
from ..expressions import Expression
from .variable_declaration import VariableDeclaration


class StaticVariableDeclaration(VariableDeclaration):
    """Represents a static variable declaration, e.g. `static a` or `static b := 1."""

    _name: str
    _assign_expr: Expression | None

    def __init__(self, name: str, assign_expr: Expression | None):
        super().__init__(variable_type="static", name=name, assign_expr=assign_expr)

File: Harpy/AST/Statements/StaticVariableDeclaration.cs
using Harpy.AST.Expressions;
using Harpy.Lexer;

namespace Harpy.AST.Statements;

/// <summary>
///     Represents a static variable declaration, e.g. <c>static a</c> or <c>static b := 1</c>.
/// </summary>
public class StaticVariableDeclaration(HarbourSyntaxToken scope, HarbourSyntaxToken name, Expression? assignment)
    : VariableDeclaration(scope, name, assignment);
